### PR TITLE
[Kotlin Cleanup] CardBrowser pt4

### PR DESCRIPTION
**Pull request by dorrin-sot**
Continued From #11460

### CODE REVIEW

File: AnkiDroid/src/main/java/com/ichi2/anki/CardBrowser.kt [hunk 1]
```diff
@@ -1027,16 +1027,12 @@ open class CardBrowser : NavigationDrawerActivity(), SubtitleListener, DeckSelec
...
-                val savedFilters: HashMap<String?, String?>
-                if (savedFiltersObj != null) {
-                    savedFilters = HashMapInit(savedFiltersObj.length())
-                    for (searchName in savedFiltersObj) {
-                        savedFilters[searchName] = savedFiltersObj.optString(searchName)
-                    }
-                } else {
-                    savedFilters = HashMapInit(0)
-                }
+                val savedFiltersObj = col.get_config("savedFilters", null as JSONObject?) ?: JSONObject()
```
Comment (Arthur-Milchior) [inline AnkiDroid/src/main/java/com/ichi2/anki/CardBrowser.kt:1030]: `val savedFiltersObj = col.get_config("savedFilters", JSONObject())!!`

**Review by Arthur-Milchior — COMMENTED** [review]
I'm done for tonight.

Thanks a lot for all those improvements. Really appreciate it.

File: AnkiDroid/src/main/java/com/ichi2/anki/CardBrowser.kt [hunk 2]
```diff
@@ -1027,16 +1027,12 @@ open class CardBrowser : NavigationDrawerActivity(), SubtitleListener, DeckSelec
...
-                if (savedFiltersObj != null) {
-                    savedFilters = HashMapInit(savedFiltersObj.length())
-                    for (searchName in savedFiltersObj) {
-                        savedFilters[searchName] = savedFiltersObj.optString(searchName)
-                    }
-                } else {
-                    savedFilters = HashMapInit(0)
-                }
+                val savedFiltersObj = col.get_config("savedFilters", JSONObject())!!
+                val savedFilters: HashMap<String?, String?> = HashMap(
```
Comment (Arthur-Milchior) [inline AnkiDroid/src/main/java/com/ichi2/anki/CardBrowser.kt:1031]: Why do you explicitly give the type here? And why nullable strings?

Comment (Arthur-Milchior) [inline AnkiDroid/src/main/java/com/ichi2/anki/CardBrowser.kt:1031]: Once JSONObject is migrated, I wonder whether we can put this function in JSONObject. It feels wrong to have it here

File: AnkiDroid/src/main/java/com/ichi2/anki/CardBrowser.kt [hunk 3]
```diff
@@ -473,12 +473,7 @@ open class CardBrowser : NavigationDrawerActivity(), SubtitleListener, DeckSelec
 
     @get:VisibleForTesting
     val lastDeckId: Long?
-        get() {
-            val state = getSharedPreferences(PERSISTENT_STATE_FILE, 0)
-            return if (!state.contains(LAST_DECK_ID_KEY)) {
-                null
-            } else state.getLong(LAST_DECK_ID_KEY, -1)
-        }
+        get() = getSharedPreferences(PERSISTENT_STATE_FILE, 0).all[LAST_DECK_ID_KEY]?.let { id -> id as Long }
```
[the remark this answers is not in the document]
Comment (dorrin-sot) [reply AnkiDroid/src/main/java/com/ichi2/anki/CardBrowser.kt]: I changed it but just for the sake of learning, can I ask why using `all` is not 
preffered?

File: AnkiDroid/src/main/java/com/ichi2/anki/CardBrowser.kt (hunk 2 above)
Comment (dorrin-sot) [reply AnkiDroid/src/main/java/com/ichi2/anki/CardBrowser.kt]: I saw that no one has taken it yet so I did 👀 #11604

**Review by Arthur-Milchior — APPROVED** [review]

**Pull request merged**